qemu/tests/watchdog.py

```diff
@@ -282,6 +282,66 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def heartbeat_test():
+        """
+        Heartbeat test for i6300esb
+        Test steps:
+        1.Start VM with "-watchdog-action pause" CLI option
+        2.Set heartbeat value and reload the i6300esb module
+        3.Trigger wathchdog action through open /dev/watchdog
+        4.Ensure watchdog_action takes effect after $heartbeat.
+        """
+
+        del_module_cmd = params["del_module_cmd"]
+        reload_module_cmd = params["reload_module_cmd"]
+        _watchdog_device_check(test, session, watchdog_device_type)
+
+        error_context.context("set heartbeat value and reload the i6300esb "
+                              "module", logging.info)
+        session.cmd(del_module_cmd)
+        heartbeat = eval(params["heartbeat"])
+        dmesg_cmd = params["dmesg_cmd"]
+        session.cmd(dmesg_cmd)
+        session.cmd_output(reload_module_cmd % heartbeat)
+        if abs(heartbeat) > 2147483647:
+            o = session.cmd_output("dmesg | grep -i 'i6300esb.*invalid'")
+            if o:
+                logging.info("Heartbeat value %s is out of range, it is "
+                             "expected." % heartbeat)
+            else:
+                test.fail("No invalid heartbeat info in dmesg.")
+        elif -2147483648 <= heartbeat < 1 or 2046 < heartbeat <= 2147483647:
+            o = session.cmd_output("dmesg | grep -i 'heartbeat=30'")
+            if not o:
+                test.fail("Heartbeat value isn't default 30 sec in dmesg, it "
+                          "should be.")
+            heartbeat = 30
+        elif 1 <= heartbeat <= 2046:
+            o = session.cmd_output("dmesg | grep -i 'heartbeat=%s'" % heartbeat)
+            if not o:
+                test.fail("Heartbeat value isn't %s sec in dmesg" % heartbeat)
```

Comment:
Would this be better:
```
max = 2147483647
if abs(heartbeat) > max:
...
elif 1 <= heartbeat <= 2046:
...
else:
...
```

Reply:
@vivianQizhu There are two values: 2147483647 and 2147483648, and I updated some codes.
(1/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.valid.i6300esb: PASS (67.15 s)
 (2/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.invalid_1.i6300esb: PASS (98.89 s)
 (3/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.invalid_0.i6300esb: PASS (91.44 s)
 (4/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.invalid_min.i6300esb: PASS (91.55 s)
 (5/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.invalid_max.i6300esb: PASS (92.93 s)
 (6/6) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.6.x86_64.myprovider.watchdog.heartbeat_test.invalid_excp.i6300esb: PASS (45.28 s)